### ISSUE TITLE
[API Proposal]: Adding headers with a method

### ISSUE THREAD

**Issue by MDshirzad** [dotnet/runtime#108148]
### Background and motivation

Firstly when I had many headers to add for my httpclient defaultHeader, I was adding them in seprate lines, then I looked for a method and I did not find that so I created a method that takes a Dictionary of string and in that method by foreach I added the keys and values to default header.
This approach made my code cleaner.

### API Proposal

```csharp
namespace System.Net.Http.System.Net.Http
{
    public static class HttpClientExtensions
    {

        public static Task AddHeader(this HttpClient httpclinet, Dictionary<string, string> dictionary)
        {
            if (dictionary == null)
                return;
            foreach (var item in dictionary)
            {
                if (!client.DefaultRequestHeaders.Contains(header.Key))
                {
                    httpclinet.DefaultRequestHeaders.Add(item.Key, item.Value);
                }
            }
            return Task.CompletedTask;


        }
    }
}
```

### API Usage

```csharp
using(var client = new HttpClient()){
var headers = new Dictionary<string,string>(){
{"X-Authorization":"abcd"},
{"X-gateway","Payment"}
};
client.AddHeader(headers);
client.post();
}
```

**Comment by MDshirzad** [comment]
@karelz

**Comment by MDshirzad** [comment]
@dotnet/ncl

**Comment by huoyaoyuan** [comment]
If all your need is to initialize `DefaultRequestHeaders` with less lines, you can just use the in-place collection initializer syntax when creating the `HttpClient`:

```csharp
var c = new HttpClient
{
    DefaultRequestHeaders =
    {
        {"X-Authorization", "abcd"},
        {"X-gateway", "Payment"}
    }
};
```

The real problem is that `HttpHeaders` doesn't have `AddRange` methods, so there's no support for bulk addition afterwards. But you can easily define you own extension method.

This proposal will likely never be considered because it doesn't operate on the correct type.

**Comment by MDshirzad** [comment]
> If all your need is to initialize `DefaultRequestHeaders` with less lines, you can just use the in-place collection initializer syntax when creating the `HttpClient`:
> 
> ```cs
> var c = new HttpClient
> {
>     DefaultRequestHeaders =
>     {
>         {"X-Authorization", "abcd"},
>         {"X-gateway", "Payment"}
>     }
> };
> ```
> 
> The real problem is that `HttpHeaders` doesn't have `AddRange` methods, so there's no support for bulk addition afterwards. But you can easily define you own extension method.
> 
> This proposal will likely never be considered because it doesn't operate on the correct type.


Consider that the time you recieve the headers from outside of your scope, for example your caller wanna send you that headers.

**Comment by antonfirsov** [comment]
Triage: we can consider an `AddRange` method on `HttpHeaders` if there are more users asking for it.

**Comment by MDshirzad** [comment]
@antonfirsov  Yeah, AddRange is better because of convention and it's place in HttpHeader👍
